ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java

```diff
@@ -493,7 +494,7 @@ public static boolean isCompactionTable(Properties tblProperties) {
    * @return true, if the parameters contains {@link AcidUtils#COMPACTOR_TABLE_PROPERTY}
    */
   public static boolean isCompactionTable(Map<String, String> parameters) {
-    return Boolean.parseBoolean(parameters.get(COMPACTOR_TABLE_PROPERTY));
+    return StringUtils.isNotBlank(parameters.get(COMPACTOR_TABLE_PROPERTY));
```

Comment:
what if the value is set to 'false'?

Reply:
It is set only by the [CompactionQueryBuilder](https://github.com/apache/hive/blob/e3751ab545370f9b252d0b4a07bc315037541a95/ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactionQueryBuilder.java), and from now on it is setting the compaction type instead of `true/false`.